modules/xmpp/SDP.js

```diff
@@ -267,26 +267,18 @@ SDP.prototype.toJingle = function(elem, thecreator) {
             }
 
             if (ssrc) {
-                // new style mapping
-                elem.c('source', { ssrc,
-                    xmlns: 'urn:xmpp:jingle:apps:rtp:ssma:0' });
-
-                // FIXME: group by ssrc and support multiple different ssrcs
-                const ssrclines = SDPUtil.findLines(this.media[i], 'a=ssrc:');
-
-                if (ssrclines.length > 0) {
-                    // eslint-disable-next-line no-loop-func
-                    ssrclines.forEach(line => {
-                        const idx = line.indexOf(' ');
-                        const linessrc = line.substr(0, idx).substr(7);
-
-                        if (linessrc !== ssrc) {
-                            elem.up();
-                            ssrc = linessrc;
-                            elem.c('source', { ssrc,
-                                xmlns: 'urn:xmpp:jingle:apps:rtp:ssma:0' });
-                        }
-                        const kv = line.substr(idx + 1);
+                const ssrcMaps = SDPUtil.parseSSRC(this.media[i]);
+
+                Object.keys(ssrcMaps).forEach(availableSsrc => {
```

Comment:
drive-by comment :-) can you please use for..in or for..of where appropriate?, we have been writing all new code with those for a while, thanks!